DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -179,5 +217,53 @@ private int GetLargestKey()
                 );
             }
         }
+
+        private async Task<int> GetEventRowNumber(int eventKey)
+        {
+            try
+            {
+                var request = sheetsService.Spreadsheets.Values.Get(spreadsheetId, "EventsMetadata!A:A");
+                var response = await request.ExecuteAsync();
+
+                if (response == null || response.Values.Count < 2)
+                {
+                    throw new EventNotFoundException($"Event key {eventKey} not recognised");
+                }
+
+                var rowNumber = response.Values
+                    .Skip(1)
+                    .Select((rowValues, rowIndex) =>
+                        (key: int.Parse((string)rowValues[0]), index: rowIndex)
+                    )
+                    .Where(row => row.key == eventKey)
+                    .Select(row => row.index + 2)
```

Comment:
Why `+ 2`??

Reply:
Yeah this is a little surprising so I've added some comments in 02e76dd 🙂 It's correcting for two things: 
 - The IEnumerable lists are 0 indexed, but Google Sheets index from 1  
 - Correct for skipping row 1, which was the header row